src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java

```diff
@@ -1,20 +1,36 @@
 package edu.oakland.production.display01;
 
 import edu.oakland.helper.admin.Satellite;
+import edu.oakland.production.display01.DisplayGpsManager;
 import edu.oakland.helper.display01.SatelliteSignalCheckRequest;
+import java.lang.IllegalArgumentException;
 
 public class DisplayGpsReceiverImplementation implements DisplayGpsReceiver {
 
-  /** 
+  private DisplayGpsManager displayGpsManager;
+
+  /**
+   * Checks whether or not passed displayGpsManager value is null.
+   *
+   * @param displayGpsManager Calls on displayGpsManager to check value for null.
+   */
+  public DisplayGpsReceiverImplementation(DisplayGpsManager displayGpsManager) {
+    if (displayGpsManager == null) {
+      throw new IllegalArgumentException("Signal strength cannot be null");
```

Comment:
This is actually checking the manager, not the signal strength.
```suggestion
      throw new IllegalArgumentException("displayGpsManager cannot be null");
```